pkg/controller/trainjob_controller.go

```diff
@@ -214,9 +275,15 @@ func isTrainJobFinished(trainJob *trainer.TrainJob) bool {
 }
 
 func (r *TrainJobReconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
-	b := ctrl.NewControllerManagedBy(mgr).
+	b := builder.TypedControllerManagedBy[reconcile.Request](mgr).
+		Named("trainjob_controller").
 		WithOptions(options).
-		For(&trainer.TrainJob{})
+		WatchesRawSource(source.TypedKind(
```

Comment:
Could `WithEventFilter` be simpler?

Reply:
Unfortunately, the TypedManager and TypedPredicate do not provide `WithEventFilter`.